lib/ast.js

```diff
@@ -146,15 +168,22 @@ function opGen(opType, arity) {
 }
 
 // A reducing function for binary operators - use with [].reduce
-function leftAssociateGen(opType) {
+// initialValue's in array are ignored (or returned for empty array).
+function leftAssociateGen(opType, initialValue) {
   function reducer(result, current) {
     if (result === undefined) {
```

Comment:
That's the way reducers are defined - the first call always has result ===
undefined.

On Tue, Sep 8, 2015 at 1:18 PM, Tom Larkworthy <notifications@github.com>
wrote:

> In lib/ast.js
> <https://github.com/firebase/bolt/pull/32#discussion_r38974492>:
>
> > @@ -146,15 +168,22 @@ function opGen(opType, arity) {
> >  }
> >
> >  // A reducing function for binary operators - use with [].reduce
> > -function leftAssociateGen(opType) {
> > +// initialValue's in array are ignored (or returned for empty array).
> > +function leftAssociateGen(opType, initialValue) {
> >    function reducer(result, current) {
> >      if (result === undefined) {
>
> Is this undefined check necessary. Seems like it could be hiding hard to
> debug bugs down the road.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/firebase/bolt/pull/32/files#r38974492>.
>



-- 
+Mike Koss <https://plus.google.com/+MikeKoss>
(425) 246-7701 (m)